crates/uv/src/commands/project/add.rs

```diff
@@ -884,10 +886,18 @@ fn augment_requirement(
     rev: Option<&str>,
     tag: Option<&str>,
     branch: Option<&str>,
+    marker: Option<MarkerTree>,
 ) -> UnresolvedRequirement {
     match requirement {
-        UnresolvedRequirement::Named(requirement) => {
+        UnresolvedRequirement::Named(mut requirement) => {
             UnresolvedRequirement::Named(uv_pypi_types::Requirement {
+                marker: match marker {
```

Comment:
should this be `and`-ed with an existing marker? is there a more elegant way to do this?

Reply:
Ah that's an interesting question. I think so?

I think the proper idiom is like `marker.map(|inner| inner.and(marker)).or(marker)`